docs/bindings.md

```diff
@@ -0,0 +1,220 @@
+retrieving available `Localization` objects.
+
+Each `document` will have its corresponding `LocalizationObserver` instance
+created automatically on startup, as `document.l10n`.
+
+## constructor
+
+Returns **[LocalizationObserver](#localizationobserver)** 
+
+## has
+
+Test if the `Localization` object with a given name already exists.
+
+```javascript
+if (document.l10n.has('extra')) {
+  const extraLocalization = document.l10n.get('extra');
+}
+```
+
+**Parameters**
+
+-   `name` **[string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)** key for the object
+
+Returns **[boolean](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** 
+
+## get
+
+Retrieve a reference to the `Localization` object by name.
+
+```javascript
+const mainLocalization = document.l10n.get('main');
+const extraLocalization = document.l10n.get('extra');
+```
+
+**Parameters**
+
+-   `name` **[string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)** key for the object
+
+Returns **Localization** 
+
+## set
+
+Sets a reference to the `Localization` object by name.
+
+```javascript
+const loc = new Localization();
+document.l10n.set('extra', loc);
+```
+
+**Parameters**
+
+-   `name` **[string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)** key for the object
+-   `value` **Localization** `Localization` object
+
+Returns **[LocalizationObserver](#localizationobserver)** 
+
+## requestLanguages
+
+Trigger the language negotation process with an array of language codes.
+Returns a promise with the negotiated array of language objects as above.
+
+```javascript
+document.l10n.requestLanguages(['de-DE', 'de', 'en-US']);
+```
+
+**Parameters**
+
+-   `requestedLangs` **[Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)&lt;[string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)>** array of requested languages
+
+Returns **[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)&lt;[Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)&lt;[string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String)>>** 
+
+## setAttributes
+
+Set the `data-l10n-id` and `data-l10n-args` attributes on DOM elements.
+L20n makes use of mutation observers to detect changes to `data-l10n-*`
+attributes and translate elements asynchronously.  `setAttributes` is
+a convenience method which allows to translate DOM elements declaratively.
+
+You should always prefer to use `data-l10n-id` on elements (statically in
+HTML or dynamically via `setAttributes`) over manually retrieving
+translations with `format`.  The use of attributes ensures that the
+elements can be retranslated when the user changes their language
+preferences.
```

Comment:
I'd like to suggest explicitly saying why we want users to set the attribute instead of just listing one example of how we may use it:

`Setting the attribute establishes a binding between the element and the entity which L20n can use for, among others, retranslation when teh user changes their language preferences`